js/src/hooks/usePagination.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Hook for handling pagination based on different pagination keys. Each key maintains their own pagination state
+ * Example of usage `const { page, setPage } = usePagination('paginator1')`
+ * setPage(2), setPage(3) ...etc
+ * Example of usage with different initial page `const { page, setPage } = usePagination('paginator2', 5)`
+ *
+ * @param {string} paginationId The key used to identify this paginator
+ * @param {number} initialPage The initial page for the paginator
```

Comment:
💅 
Suggest indicating it is optional and has a default value.

Reply:
Thanks. Done  here https://github.com/woocommerce/google-listings-and-ads/pull/1176/commits/f7bb1266d37a97d902e7539f74623f6348f582df